packages/frontend-app-api/src/tree/instantiateAppNodeTree.ts

```diff
@@ -46,26 +46,26 @@ function resolveInputData(
 }
 
 function resolveInputs(
+  id: string,
   inputMap: AnyExtensionInputMap,
   attachments: ReadonlyMap<string, AppNode[]>,
 ): ResolvedExtensionInputs<AnyExtensionInputMap> {
   const undeclaredAttachments = Array.from(attachments.entries()).filter(
     ([inputName]) => inputMap[inputName] === undefined,
   );
-  // TODO: Make this a warning rather than an error
-  if (undeclaredAttachments.length > 0) {
-    throw new Error(
-      `received undeclared input${
-        undeclaredAttachments.length > 1 ? 's' : ''
-      } ${undeclaredAttachments
-        .map(
-          ([k, exts]) =>
-            `'${k}' from extension${exts.length > 1 ? 's' : ''} '${exts
-              .map(e => e.spec.id)
-              .join("', '")}'`,
-        )
-        .join(' and ')}`,
-    );
+
+  if (process.env.NODE_ENV !== 'production') {
+    for (const [name, nodes] of undeclaredAttachments) {
+      const pl = nodes.length > 1;
+      // eslint-disable-next-line no-console
+      console.warn(
+        `The extension${pl ? 's' : ''} '${nodes
+          .map(n => n.spec.id)
+          .join("', '")}' ${
+          pl ? 'are' : 'is'
+        } attached to the input '${name}' of '${id}', but the extension '${id}' noes not declare a '${name}' input`,
```

Comment:
Bit too much duplication; just "but it had no such input (candidates are a, b, c)" would be more helpful and have an acceptable level of risk of misunderstanding due to syftningsfel